components/logins/src/db.rs

```diff
@@ -1235,4 +1268,108 @@ mod tests {
         db.delete(login.guid_str()).unwrap();
         assert!(&unique_login_check.is_ok())
     }
+
+    fn check_matches(db: &LoginDb, query: &str, expected: &[&str]) {
+        let mut results = db
+            .get_by_base_domain(query)
+            .unwrap()
+            .into_iter()
+            .map(|l| l.hostname)
+            .collect::<Vec<String>>();
+        results.sort();
+        let mut sorted = expected.to_owned();
+        sorted.sort();
+        assert_eq!(sorted, results);
+    }
+
+    fn check_good_bad(
+        good: Vec<&str>,
+        bad: Vec<&str>,
+        good_queries: Vec<&str>,
+        zero_queries: Vec<&str>,
+    ) {
+        let db = LoginDb::open_in_memory(Some("testing")).unwrap();
+        for h in good.iter().chain(bad.iter()) {
+            db.add(Login {
+                hostname: (*h).into(),
+                http_realm: Some((*h).into()),
+                password: "test".into(),
+                ..Login::default()
+            })
+            .unwrap();
+        }
+        for query in good_queries {
+            check_matches(&db, query, &good);
+        }
+        for query in zero_queries {
+            check_matches(&db, query, &[]);
+        }
+    }
+
+    #[test]
+    fn test_get_by_base_domain_invalid() {
+        check_good_bad(
+            vec!["https://example.com"],
+            vec![],
+            vec![],
+            vec!["invalid query"],
+        );
+    }
+
+    #[test]
+    fn test_get_by_base_domain() {
+        check_good_bad(
+            vec![
+                "https://example.com",
+                "https://www.example.com",
+                "http://www.example.com",
+                "http://www.example.com:8080",
+                "http://sub.example.com:8080",
+                "https://sub.example.com:8080",
```

Comment:
Can you include an ftp:// test case too to remind people to think about non-http schemes.